routing/index_graph_starter.cpp

```diff
@@ -176,9 +255,45 @@ void IndexGraphStarter::FindPointsWithCommonFeature(Joint::Id jointId0, Joint::I
   CHECK(found, ("Can't find common feature for joints", jointId0, jointId1));
 }
 
+void IndexGraphStarter::AddAdditionalZeroEdges(FakeJoint const & fp, EndPointType endPointType)
+{
+  CHECK(!m_graph.IsFakeFeature(fp.m_point.GetFeatureId()), ());
+
+  bool const isJoint = fp.BelongsToGraph();
+
+  vector<DirectedEdge> edges;
+  if (isJoint)
+  {
+    m_graph.GetEdgeList(fp.m_jointId, endPointType == EndPointType::Start /* is outgoing */,
+                        true /* graphWithoutRestrictions */, edges);
+  }
+  else
+  {
+    m_graph.GetIntermediatePointEdges(fp.m_point, true /* graphWithoutRestrictions */, edges);
+  }
+
+  for (DirectedEdge const & edge : edges)
+  {
+    m_graph.ForEachEdgeMappingNode(edge, [&](DirectedEdge const & e){
+      if (edge == e)
+        return;
+
+      switch (endPointType) {
+      case EndPointType::Start:
+        AddZeroLengthOnewayFeature(isJoint ? edge.GetFrom() : edge.GetTo(),
```

Comment:
I'm copy paste my comments and questions about A* potential which I wrote above:
"
I didn't quite understand the word "potential" in this context. Is your question if the A Star heuristic is overestimated or not?

Here I connect two joints which have the same origin joint and correspond two points with exactly the same coords with a zero length edge. CalcHeuristic(e.GetTo(), e.GetFrom()) == 0. So the heuristic is admissible.

Start joint (or if the start is not a joint a joint closest to the start road point if go along to the edge) is connected with joints which originate (while adding restrictions process) from the start joint with zero length edge (if any). More or less the same happens with finish. This implemented in IndexGraphStarter::AddAdditionalZeroEdges().

Could you please explain which potential problem you see here?
"